provider/testdata/src/lib.rs

```diff
@@ -92,7 +92,7 @@ pub mod versions {
     /// # Examples
     ///
     /// ```
-    /// assert_eq!("icu4x/2023-02-24/72.x", icu_testdata::versions::icu_tag());
+    /// assert_eq!("icu4x/2023-03-22a/72.x", icu_testdata::versions::icu_tag());
```

Comment:
Nit: I'd prefer for the tag change to be its own standalone PR (I've given similar feedback to others in the past)

Reply:
I just remembered: in order for `ci-job-testdata` to pass, the tag needs to be updated because it refers to the data from the latest tag when running `cargo make testdata`. So if we want the latest version tag to be changed in a separate PR, then merging this PR will require overriding the fact that the relevant CI check will be failing (and that the follow on CI check `full-datagen` won't run as a result, too).

Reply:
You can include the tag update in this PR but also pull it out, merge the other PR, and then update this PR to the latest main, which should be a small change. But I'll leave it up to you if you want to do this. It's a preference of mine but not super strong since this is only an additive change.

Reply:
Done.